src/QsCompiler/Core/TransformationOptions.fs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.QsCompiler.Transformations.Core
+
+
+/// Used to configure the behavior of the default implementations for transformations.
+type TransformationOptions = {
+    /// Disables the transformation at the transformation root, 
+    /// meaning the transformation won't recur into leaf nodes or subnodes. 
+    Disable : bool
+    /// Indicates that the transformation is used to walk the syntax tree, but does not modify any of the nodes. 
+    /// If set to true, the nodes will not be rebuilt during the transformation. 
+    /// Setting this to true constitutes a promise that the return value of all methods will be ignored. 
+    DisableRebuild : bool
```

Comment:
I wonder if something like `ReadOnly` would be a better name for this option.

Reply:
I think it might be more confusing, since all data structures themselves are immutable?

Reply:
That makes sense. My second choice would be to change the default value to true and rename this option to `RebuildTree`, e.g., to avoid double-negatives like "disable rebuild is false => rebuild is enabled," but it's not a big deal either way.

Reply:
changed to Rebuild